File: context_tools/stories/templates/ts/{epic-verb-noun}/{sub-epic-verb-noun}/{story-verb-noun}/{story_verb_noun}_test_helper.server.ts
/** Tier: server - {StoryVerbNoun}Helper backed by Supertest against the real route. */
import { describe } from "vitest";
import { create{StoryVerbNoun}Story, type {StoryVerbNoun}Helper } from "./{story_verb_noun}_story";

class ServerHelper implements {StoryVerbNoun}Helper {
  givenPrecondition(): void | Promise<void> {
    throw new Error("not implemented: givenPrecondition");
  }
  whenAction(): void | Promise<void> {
    throw new Error("not implemented: whenAction");
  }
  thenOutcome(): void | Promise<void> {
    throw new Error("not implemented: thenOutcome");
  }
}

describe("tier: server", () => {
  create{StoryVerbNoun}Story(new ServerHelper());
});

File: context_tools/stories/templates/py/{epic-verb-noun}/{sub-epic-verb-noun}/{story-verb-noun}/{story_verb_noun}_test_helper.server.py
# ---
# fidelity: [engineering]
# artifact: [test-helper]
# format: py
# ---
#
# Tier: server - {EpicVerbNoun}Helper backed by the real HTTP route + test DB.
# Real: domain + repository + test DB. Stubbed: nothing.

"""Server tier test-helper for `{Story Name}`."""

from __future__ import annotations

from {story_verb_noun}_story import create_{story_verb_noun}_story


class ServerHelper:
    def given_precondition(self) -> None:
        raise NotImplementedError("not implemented: given_precondition")

    def when_action(self) -> None:
        raise NotImplementedError("not implemented: when_action")

    def then_outcome(self) -> None:
        raise NotImplementedError("not implemented: then_outcome")


globals().update(create_{story_verb_noun}_story(ServerHelper()))
